### PR TITLE
[Resource] Prevent extra translation lookup if locale is same as fallback locale

### PR DESCRIPTION
**Pull request by dannyvw**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Component/Resource/Model/TranslatableTrait.php [hunk 1]
```diff
@@ -78,11 +78,13 @@ public function getTranslation($locale = null)
             return $translation;
         }
 
-        $fallbackTranslation = $this->translations->get($this->fallbackLocale);
-        if (null !== $fallbackTranslation) {
-            $this->translationsCache[$this->fallbackLocale] = $fallbackTranslation;
+        if ($locale !== $this->fallbackLocale) {
+            $fallbackTranslation = $this->translations->get($this->fallbackLocale);
```
Comment (pamil) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: I'd add that caching part before getting a translation from the collection:

```php
if (isset($this->translationsCache[$this->fallbackLocale])) {
    return $this->translationsCache[$this->fallbackLocale];
}
```

Comment (dannyvw) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: @pamil You mean this?

```
        if ($locale !== $this->fallbackLocale) {
            if (isset($this->translationsCache[$this->fallbackLocale])) {
                 return $this->translationsCache[$this->fallbackLocale];
            }

            $fallbackTranslation = $this->translations->get($this->fallbackLocale);
            if (null !== $fallbackTranslation) {
                $this->translationsCache[$this->fallbackLocale] = $fallbackTranslation;
                return $fallbackTranslation;
            }
        }
```

Comment (pamil) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: Yes, IIRC `$this->translations->get(...)` almost always results in a query, cache was introduced to solve it. :)

Comment (dannyvw) [inline src/Sylius/Component/Resource/Model/TranslatableTrait.php]: PR is updated

**Comment by pamil** [comment]
Thanks @dannyvw! 🎉

**Pull request merged**